src/renderers/dom/client/utils/createObjectWithProto.js

```diff
@@ -0,0 +1,33 @@
+/**
```

Comment:
@sebmarkbage was the one who proposed using an ES6 class here, I'll let him make the final call here.